### ISSUE TITLE
Crash with AWS provider creating aws_instance

### ISSUE THREAD

**Issue by rberlind** [hashicorp/terraform#19167]
### Terraform Version
```
$terraform -v
Terraform v0.12.0-alpha1
+ provider.aws v1.40.0-5-g4f24bc8b7
```

### Terraform Configuration Files

```hcl
terraform {
  required_version = ">= 0.12.0"
}

provider "aws" {
  region = var.aws_region
}

resource "aws_instance" "ubuntu" {
  ami                         = var.ami_id
  instance_type               = var.instance_type
  availability_zone           = format("%sb", var.aws_region)
  associate_public_ip_address = "true"
  tags = {
    Name  = var.name
    owner = "roger@hashicorp.com"
    ttl   = "-1"
  }
}
```

### Debug Output
https://gist.github.com/rberlind/fbac8ad21e592649e939fa89b8874c6e

### Crash Output
https://gist.github.com/rberlind/b5b8dc06fc289725a8a2fe6b0801b3ea

### Expected Behavior
Apply should have worked.
### Actual Behavior
Crash.

### Steps to Reproduce
1. terraform init
2. terraform apply

### Additional Context
The panic message I saw in the crash ("can't use ElementIterator on null value") is the same as in https://github.com/hashicorp/terraform/issues/19152 even though I used the AWS provider and that issue used the Google provider.

Note that I did use the AWS provider that was bundled with the Terraform 0.12 alpha-1

### References
Possibly https://github.com/hashicorp/terraform/issues/19152

**Comment by rberlind** [comment]
I should note that I was able to use the AWS provider to create a VPC with some simple code:
```hcl
provider "aws" {
  region = "us-east-1"
}

resource "aws_vpc" "main" {
  cidr_block = "10.0.0.0/16"
  tags = {
    Name = "roger-test-0.12"
  }
}
```

**Comment by mitchellh** [comment]
Noting that this is the same crash as #19152.

**Comment by rberlind** [comment]
Strangely, my AWS example works if I add a credit_specification block at the end:
```hcl
terraform {
  required_version = ">= 0.12.0"
}

provider "aws" {
  region = var.aws_region
}

resource "aws_instance" "ubuntu" {
  ami           = var.ami_id
  instance_type = var.instance_type
  availability_zone = format("%sb", var.aws_region)
  tags = {
    Name  = var.name
    owner = "roger@hashicorp.com"
    ttl   = "-1"
  }
  credit_specification {
    cpu_credits = "unlimited"
  }
}
```

I discovered this accidentally while implementing this [example](https://www.terraform.io/docs/providers/aws/r/instance.html#example). That example works with either the credit_specification or network_interface block or both of them.

**Comment by rberlind** [comment]
Thanks @mitchellh

**Comment by rberlind** [comment]
I can verify this is fixed in 0.12 alpha-2
